app/views.py

```diff
@@ -90,61 +94,88 @@ def progressreport():
-#        temp = set_code(6)
-#        while Group.query.filter_by(classCode=temp).first() is not None:
-#            temp = set_code(6)
-#        group = Group(name=form.name.data, classCode=temp)
-#        db.session.add(group)
-#        db.session.commit()
-#        return redirect(url_for('createclasssuccess'))
-#    return render_template('createclass.html', title=' | Create Class', form=form)
-
-@app.route('/createclasssuccess', methods=['GET'])
-def createclasssuccess():
+@app.route('/class', methods=['POST'])
+def createclass():
+    """Renders the create class page for educators."""
+    if not current_user.check_educator():
+        return render_template('error404.html'), 404
+    classForm = CreateName(prefix='class')
+    quizForm = CreateName(prefix='quiz')
+    image_file = url_for('static', filename='resources/images/profile_pics/' + current_user.image_file)
+    if classForm.validate_on_submit():
+        group = Group(name=classForm.name.data)
+        set_class_code(group)
+        group.users.append(current_user)
+        db.session.add(group)
+        db.session.commit()
+        return redirect(url_for('createclasssuccess', groupID=group.id))
+    return render_template('dashboard.html', image_file=image_file, classForm=classForm, quizForm=quizForm)
+
+@app.route('/class/<int:groupID>/success')
+def createclasssuccess(groupID):
     """Renders the create class was a success page for educators."""
-    return render_template('createclasssuccess.html', title=' | Create Class')
-
-#@app.route('/createquiz', methods=['GET', 'POST'])
-#def createquiz():
-#    """Renders the create quiz page for educators."""
-#    form = CreateName()
-#    if form.validate_on_submit():
-#        #quiz = Quiz(name=form.name)
-#        #db.session.add(quiz)
-#        #db.session.commit()
-#        return redirect(url_for('createqn'))
-#    return render_template('createquiz.html', title=' | Create Quiz', form=form)
-
-@app.route('/createnewquestion', methods=['GET', 'POST'])
-def createqn():
+    group = validate_group_link(groupID)
+    return render_template('createclasssuccess.html', title=' | Create Class', group=group)
+
+@app.route('/quiz/<int:quizID>')
+def preview_quiz(quizID):
+    """Renders the create class was a success page for educators."""
+    if not current_user.check_educator():
+        return render_template('error404.html'), 404
+    quiz = validate_quiz_link(quizID)
+    questions = get_question_quiz(quiz)
+    return render_template('classquizzes.html', title=' | Create Class', questions=questions)
+
+@app.route('/quiz', methods=['POST'])
+def createquiz():
+    """Renders the create quiz page for educators."""
+    if not current_user.check_educator():
+        return render_template('error404.html'), 404
+    classForm = CreateName(prefix='class')
+    quizForm = CreateName(prefix='quiz')
+    image_file = url_for('static', filename='resources/images/profile_pics/' + current_user.image_file)
```

Comment:
Is this supposed to be the user image upload? Is not a required field, and this adds the educator's profile image. Not sure why you have this?

Reply:
This was yours. It's the user's own profile image.